### PR TITLE
Add Kubeflow 1.7 Release Page

### PR DESCRIPTION
**Pull request by DomFleischmann**
Add release page for Kubeflow 1.7

### CODE REVIEW

File: content/en/docs/releases/kubeflow-1.7.md [hunk 1]
```diff
@@ -0,0 +1,272 @@
++++
+title = "Kubeflow 1.7"
+description = "Information about the Kubeflow 1.7 release"
+weight = 98
```
Comment (annajung) [inline content/en/docs/releases/kubeflow-1.7.md:4]: The weight needs to be lower than the 1.6 release page to allow the 1.7 release page to be displayed first

currently, https://deploy-preview-3476--competent-brattain-de2d6d.netlify.app/docs/releases/ shows 1.6 first

File: content/en/docs/releases/kubeflow-1.7.md [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+## Kubeflow 1.7.0
+
+<div class="table-responsive">
+<table class="table table-bordered">
+  <tbody>
+    <tr>
+      <th class="table-light">Release Date</th>
+      <td>
+        2023-03-15
```
Comment (annajung) [inline content/en/docs/releases/kubeflow-1.7.md:15]:
```suggestion
        2023-03-29
```

File: content/en/docs/releases/kubeflow-1.7.md [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+      </td>
+    </tr>
+    <tr>
+      <th class="table-light">Manifests</th>
+      <td>
+        <b>Release:</b> 
+          <a href="https://github.com/kubeflow/manifests/releases/tag/v1.7.0">v1.7.0</a>
+        <br>
+        <b>Branch:</b>
+          <a href="https://github.com/kubeflow/manifests/tree/v1.7-branch">v1.6-branch</a>
```
Comment (annajung) [inline content/en/docs/releases/kubeflow-1.7.md:38]:
```suggestion
          <a href="https://github.com/kubeflow/manifests/tree/v1.7-branch">v1.7-branch</a>
```

File: content/en/docs/releases/kubeflow-1.7.md [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+        <td>Training Operator</td>
+        <td>
+          <a href="https://github.com/kubeflow/training-operator/releases/tag/v1.6.0">v1.6.0</a>
+        </td>
+      </tr>
+  </tbody>
+</table>
+</div>
+
+### Dependency Versions (Manifests)
```
[the remark this answers is not in the document]
Comment (DomFleischmann) [reply content/en/docs/releases/kubeflow-1.7.md:185]: It wasn't included in any other release page, so for now I will leave it out.

**Comment by annajung** [comment]
/lgtm

cc @zijianjoy to review

**Comment by zijianjoy** [comment]
/lgtm
/approve

Thank you for the release note!

**Pull request merged**